src/sctools/bam.py

```diff
@@ -130,34 +193,56 @@ def tag(self, output_bam_name, tag_generators):
 
 
 def split(in_bam, out_prefix, *tags, approx_mb_per_split=1000, raise_missing=True):
-    """
-    split a bam file into subfiles by tag, calculating the number of splits so that the chunks are
-    approximately approx_mb_per_split
-
-    :param str in_bam: input bam file
-    :param str out_prefix:  prefix for all output files; output will be named as prefix_n where n
-      is an integer equal to the chunk number.
-    :param list tags: the bam tags to split on. The tags are checked in order, and sorting is done
-      based on the first identified tag. Further tags are only checked if the first tag is missing.
-      This is useful in cases where sorting is executed over a corrected barcode, but some records
-      only have a raw barcode.
-    :param float approx_mb_per_split: the target file size for each chunk in mb
-    :param raise_missing: default=True, if True, raise a RuntimeError if a record is encountered
-      without a tag. Else silently discard the record.
-
-    :return [str]: output file names
+    """split `in_bam` by tag into files of `approx_mb_per_split`
+
+    Parameters
+    ----------
+    in_bam : str
+        Input bam file.
+    out_prefix : str
+        Prefix for all output files; output will be named as prefix_n where n is an integer equal
+        to the chunk number.
+    tags : list
+        The bam tags to split on. The tags are checked in order, and sorting is done based on the
+        first identified tag. Further tags are only checked if the first tag is missing. This is
+        useful in cases where sorting is executed over a corrected barcode, but some records only
+        have a raw barcode.
+    approx_mb_per_split : float
+        The target file size for each chunk in mb
+    raise_missing : bool, optional
+        if True, raise a RuntimeError if a record is encountered without a tag. Else silently
+        discard the record (default = True)
+
```

Comment:
It would be better to have a consistent annotation among the repo, either PEP484 to all funcs, or not. But it can be gradually done, no need to be finished in this PR.

Reply:
Sorry, missed this one 👍